PWGDQ/Core/MCSignal.h

```diff
@@ -147,17 +147,36 @@ bool MCSignal::CheckProng(int i, bool checkSources, const U& mcStack, const T& t
         }
       }
     }
-    // make sure that a mother exists in the stack before moving one generation further in history
-    if (!currentMCParticle.has_mothers() && j < fProngs[i].fNGenerations - 1) {
-      return false;
-    }
-    if (currentMCParticle.has_mothers() && j < fProngs[i].fNGenerations - 1) {
+
+    // if checking back in time: look for mother
+    // else (checking in time): look for daughter
+    if (!fProngs[i].fCheckGenerationsInTime) {
+      // make sure that a mother exists in the stack before moving one generation further in history
+      if (!currentMCParticle.has_mothers() && j < fProngs[i].fNGenerations - 1) {
+        return false;
+      }
       /*for (auto& m : mcParticle.mothers_as<aod::McParticles_001>()) {
         LOGF(debug, "M2 %d %d", mcParticle.globalIndex(), m.globalIndex());
       }*/
-      currentMCParticle = currentMCParticle.template mothers_first_as<U>();
-      // currentMCParticle = mcStack.iteratorAt(currentMCParticle.mothersIds()[0]);
-      // currentMCParticle = currentMCParticle.template mother0_as<U>();
+      if (currentMCParticle.has_mothers() && j < fProngs[i].fNGenerations - 1) {
+        currentMCParticle = currentMCParticle.template mothers_first_as<U>();
+        // currentMCParticle = mcStack.iteratorAt(currentMCParticle.mothersIds()[0]);
+        // currentMCParticle = currentMCParticle.template mother0_as<U>();
+      }
+    } else {
+      // make sure that a daughter exists in the stack before moving one generation younger
+      if (!currentMCParticle.has_daughters() && j < fProngs[i].fNGenerations - 1) {
+        return false;
+      }
+      if (currentMCParticle.has_daughters() && j < fProngs[i].fNGenerations - 1) {
+        const auto& daughtersSlice = currentMCParticle.template daughters_as<U>();
+        for (auto& d : daughtersSlice) {
+          if (abs(d.pdgCode()) == fProngs[i].fPDGcodes[j + 1]) {
+            currentMCParticle = d;
+            break;
```

Comment:
Here you check the absolute of the PDG code, however, the user has the option to choose between checking both charges (using the abs of the PDG code), or not. So the latter situation is overwritten here. Could you fix it? Ideally one should use here the MCProng::TestPDG() since that will take into account automatically these options.